app/src/androidTest/java/org/wikipedia/pages/SearchBarPage.kt

```diff
@@ -0,0 +1,58 @@
+package org.wikipedia.pages
+
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.action.ViewActions.*
+import androidx.test.espresso.matcher.ViewMatchers.*
+import org.hamcrest.Matchers.allOf
+import org.wikipedia.R
+
+
+
+class SearchBarPage : BasePage() {
+
+    private var text: String = ""
+    private val searchTextInput = withId(R.id.search_src_text)
+    private val pageList = withId(R.id.page_list_item_title)
+    private val recentSearchesDeleteButton = withId(R.id.recent_searches_delete_button)
+    private val yesButton = withText(R.string.dialog_message_clear_history_yes)
+    private val emptyImage = withId(R.id.search_empty_image)
+    private val resultText = hasDescendant(withText(text))
+
+
+    fun typeTextSearch(text: String) {
+        onView(searchTextInput).perform(typeText(text))
+    }
+
+    fun isDisplayedElement(): Boolean {
```

Comment:
This method will almost always return true. The `resultText` matcher will look for a descendant with empty text - in view hierarchy there might be a lot of such views however, you `viewExists` method will catch the `AmbigiousViewMatcherException`. Is that what you want to achieve here?

If you want to get a matcher based on provided text, I would create a custom function for that:
```
fun searchResultForText(searchQuery): Matcher<View> = hasDescendant(withText(searchQuery))
```

Place it together with other Matchers - it doesn't really matter if you create them via variables (as you do) or you are using a functions for that

Reply:
we have to discuss it during the meeting :(